src/Optimization/UI/AxisymmetricOptimizationUI.jl

```diff
@@ -0,0 +1,546 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+module AxisymmetricOptimizationUI
+
+const INTERACTIVE_HTML = false
+
+using OpticSim
+using Blink, WebIO, Interact, Interact.CSSUtil, DataFrames, CSV, TableView
+using ..BlinkUtils
+using ..AxisymmetricOptimization
+
+#
+import Makie
+import Makie.AbstractPlotting
+import Makie.AbstractPlotting.MakieLayout
+if (INTERACTIVE_HTML)
+    import WGLMakie
+else
+    import GLMakie
+end
+
+
+function rg_showtable_sync!(w, schema, names, types, rows, coldefs, tablelength, id, options)
+    # RG - disable sorting for all columns
+    for c in coldefs
+        c[:sortable] = false
+    end
+      
+    options[:rowData] = TableView.JSONText(TableView.table2json(schema, rows, types))
+    license = get(ENV, "AG_GRID_LICENSE_KEY", nothing)
+    handler = @js function (RowNumberRenderer, agGrid)
+        @var gridOptions = $options
+        @var el = document.getElementById($id)
+        gridOptions.components = Dict(
+            "rowNumberRenderer" => RowNumberRenderer
+        )
+        if $(license !== nothing)
+            agGrid.LicenseManager.setLicenseKey($license)
+        end
+        this.table = @new agGrid.Grid(el, gridOptions)
+        gridOptions.columnApi.autoSizeAllColumns()
+    end
+    onimport(w, handler)
+end
+
+function run()
+
+    @eval begin
+        # this function is needed to allow the visualization scene to be displayed inside a blink window
+        function Makie.display(obj)
+            # @info "RG: $obj"
+            return obj
+        end
+
+        function Vis.scene(resolution = (1000, 1000))
+            # @info "RG: Vis.Scene Replacement"
+            scene, layout = MakieLayout.layoutscene(resolution = resolution)
+            Vis.set_current_main_scene(scene)
+            lscene = layout[1, 1] = MakieLayout.LScene(scene, scenekw = (camera = Makie.cam3d_cad!, axis_type = Makie.axis3d!, raw = false))
+            Vis.set_current_3d_scene(lscene)
+            return scene, lscene
+        end
+
+        function TableView._showtable_sync!(w, schema, names, types, rows, coldefs, tablelength, id, options)
+            # RG - disable sorting for all columns
+            for c in coldefs
+                c[:sortable] = false
+            end
+            rg_showtable_sync!(w, schema, names, types, rows, coldefs, tablelength, id, options)  
+        end
```

Comment:
Is there no other way to do this other than overwriting a private TableView function? Also you have the code for "RG - disable sorting for all columns" twice (it's also in rg_show_table_sync above). Is this necessary?